### PR TITLE
Provide (experimental) support for mkdocs documentation

### PR DESCRIPTION
**Pull request by jaimergp**
## Description

This is an attempt at using `mkdocs` + `material` + `mkdocstrings` work as a replacement for ReadTheDocs.

URL: http://openkinome.github.io/kinoml/

## Tasklist

- [x] Remove `mkdocs` branch from Docs CI events

## Gotchas

- There's no standard `autodoc` replacement; instead there are many which cannot be considered very mature yet. One striking difference might be that since `mkdocs` is `markdown` based, docstrings purposedly avoid using `numpydoc` docstrings because they resemble RST too much. I don't see this as a problem for new projects, but might be something to discuss. There are other plugins in the making that will add support for `numpydoc` but none robust yet.
- We won't deploy to `readthedocs` but to `github.io` automatically, thanks to GH Actions.
- It looks too beautiful!

### CODE REVIEW

File: kinoml_env.yml [hunk 1]
```diff
@@ -0,0 +1,197 @@
+name: kinoml
```
Comment (jchodera) [inline kinoml_env.yml]: Do we want to pull in this frozen environment? I think we want our environment YAML files to be minimal?

File: mkdocs_env.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: mkdocs
```
Comment (jchodera) [inline mkdocs_env.yml]: I think we want our mkdocs environments to just be minimal, right? Not pinned versions of everything?

Comment (jaimergp) [reply mkdocs_env.yml]: Oh, oh! This shouldn't be here at all. I used to debug some version-drive misbehavior. Relevant packages are [here](https://github.com/openkinome/kinoml/pull/10/files/fb0c7342e6111f8d89ac245d9c147261945f34bb#diff-989fa9fc279181551c489ab45b745663R14). I still need to pin specific versions due to [this issue](https://github.com/pawamoy/mkdocstrings/issues/47), but that only affects `mkdocs`. I will remove all this dirt (and merge via squashing) before merge. Sorry!

File: kinoml/smth.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    You can use more lines.
+
+        This is code block,
+        as usual.
+
+    ```python
+    s = "This is a Python code block :)"
+    ```
+
+    Arguments:
```
Comment (jaimergp) [inline kinoml/smth.py]: @jchodera, are we sure we want to go and use this syntax for our library? I wouldn't mind at all, but we have to make the decision now before I start writing too many docstrings.

**Comment by jaimergp** [comment]
I just discovered all the [Material theme extensions](https://squidfunk.github.io/mkdocs-material/extensions/admonition/) (which I promptly enabled) and I am _deeply_ [in love](https://openkinome.github.io/kinoml/howtodocs/#how-to-write-docs-with-mkdocs).

**Pull request merged**